org.scala-ide.sdt.core/src/org/scalaide/core/internal/jdt/model/ScalaJavaMapper.scala

```diff
@@ -217,34 +217,41 @@ trait ScalaJavaMapper extends InternalCompilerServices with ScalaAnnotationHelpe
       javaDescriptor(tpe).replace('/', '.')
   }
 
-  import genBCode.bTypes._
-
   /** Return the descriptor of the given type. A typed descriptor is defined
    *  by the JVM Specification Section 4.3 (http://docs.oracle.com/javase/specs/vms/se7/html/jvms-4.html#jvms-4.3)
    *
    *  Example:
-   *   javaDescriptor(Array[List[Int]]) == "[Lscala/collection/immutable/List;"
+   *   toJavaDescriptor(Array[List[Int]]) == "[Lscala/collection/immutable/List;"
    */
-  private def javaDescriptor(bt: BType): String = {
-    import Signature._
-    (bt: @unchecked) match {
-      case BOOL             => C_BOOLEAN.toString
-      case BYTE             => C_BYTE.toString
-      case SHORT            => C_SHORT.toString
-      case CHAR             => C_CHAR.toString
-      case INT              => C_INT.toString
-      case UNIT             => C_VOID.toString
-      case LONG             => C_LONG.toString
-      case FLOAT            => C_FLOAT.toString
-      case DOUBLE           => C_DOUBLE.toString
-      case ClassBType(cls)  => s"L$cls;"
-      case ArrayBType(elem) => s"[${javaDescriptor(elem)}"
+  private def toJavaDescriptor(tpe: Type): String = {
```

Comment:
@lrytz Could you have a look at this function? Maybe you can tell us if it is a good replacement for the function above. We want to get rid of `BType` but it has quite a complex implementation and therefore we do not know if we covered all cases.